adi/ad4630.py

```diff
@@ -166,3 +170,70 @@ def calibscale(self):
         def calibscale(self, calibscale):
             """Set calibration scale value."""
             self._set_iio_attr(self.name, "calibscale", False, calibscale, self._ctrl)
+
+
+class adaq42xx(ad4630):
+
+    """ ADAQ4224 is a 24-bit precision SAR ADC data acquisition module """
+
+    def __init__(self, uri="", device_name="adaq4224"):
+
+        context_manager.__init__(self, uri, self._device_name)
+
+        compatible_parts = ["adaq4224", "adaq4216", "adaq4220"]
+
+        if device_name not in compatible_parts:
+            raise Exception(
+                "Not a compatible device: "
+                + str(device_name)
+                + ". Please select from "
+                + str(compatible_parts)
+            )
+        else:
+            self._ctrl = self._ctx.find_device(device_name)
+            self._rxadc = self._ctx.find_device(device_name)
+
+        self.__init_channels__(self)
```

Comment:
This constructor is almost identical to the parent. Can you move "compatible_parts" to a class prop and set it uniquely for parent and child

Reply:
Sure, made the child class use parent's constructor on V2.